dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/ApiKeyUtilityTest.kt

```diff
@@ -0,0 +1,107 @@
+package org.dataland.datalandbackend.services
+
+import org.dataland.datalandbackendutils.apikey.ApiKeyUtility
+import org.dataland.datalandbackendutils.apikey.ParsedApiKey
+import org.dataland.datalandbackendutils.exceptions.ApiKeyFormatException
+import org.dataland.datalandbackendutils.utils.EncodingUtils
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class ApiKeyUtilityTest {
+
+    private val apiKeyUtility = ApiKeyUtility()
+
+    private val testKeycloakUserId = "c5ef10b1-de23-4a01-9005-e62ea226ee83"
+    private val testApiKeyBase64EncodedKeycloakUserId = "YzVlZjEwYjEtZGUyMy00YTAxLTkwMDUtZTYyZWEyMjZlZTgz"
+    private val testApiKeySecret = "da030a8c290f43a022bdb3da59f6ee4d7b5e290997fbdb3a22f1a7ce5a1a51c87f7666a3476a950b"
+    private val testApiKeyCrc32Value = "465063745"
+
+    private fun getCrc(keycloakUserIdBase64Encoded: String, apiKeySecret: String): String {
+        val parsedApiKeyWithoutCrc32Value = keycloakUserIdBase64Encoded + "_" + apiKeySecret
+
+        return EncodingUtils.calculateCrc32Value(
+            parsedApiKeyWithoutCrc32Value.toByteArray()
+        ).toString()
+    }
+
+    private fun prevalidateBrokenApiKeysAndAssertThrownMessages(
+        mapOfBrokenApiKeysAndExpectedMessages:
+            Map<String, String>
+    ) {
+        mapOfBrokenApiKeysAndExpectedMessages.forEach { (brokenApiKey, expectedMessage) ->
+            val thrown = assertThrows<ApiKeyFormatException> {
+                apiKeyUtility.parseApiKey(brokenApiKey)
+            }
+            assertEquals(
+                expectedMessage,
+                thrown.message
+            )
+        }
+    }
+
+    @Test
+    fun `check if prevalidation passes for a correct api key`() {
+        val apiKey = testApiKeyBase64EncodedKeycloakUserId + "_" + testApiKeySecret + "_" + testApiKeyCrc32Value
+        val parsedApiKey = apiKeyUtility.parseApiKey(apiKey)
+        val expectedParsedApiKey = ParsedApiKey(
+            testKeycloakUserId, testApiKeySecret
+        )
+        assertEquals(expectedParsedApiKey, parsedApiKey)
+    }
+
+    @Test
+    fun `check if exception thrown if the provided api key does not have the exact number of delimiters`() {
```

Comment:
der Name und was im Test passiert passen nicht zusammen (nur einer der beiden checks ist im Namen enthalten)

Reply:
Meine Absicht hier war, auf einen case abzuzielen und zu testen (und zwar dass die Anzahl der Delimiter falsch ist), dies aber in verschiedenen Varianten, in denen das auftreten kann, zu tun.
Einmal die Variante, dass ein "_" zu viel ist, der key aber ansonsten vernünftig ausschaut.
Und einmal die Variante mit völligem nonsense im api key.
Daher wird in der Überschrift nur ein Case genannt.

Ich bin aber offen für Vorschläge. Vllt den "nonsense" case in einen iegenen Test stecken? Wegschmeißen?

Reply:
Die Varianten teilen sich auch die exception message, weshalb ich es in einen Test gesteckt habe.

Reply:
Achso es ging dir um die Fälle ein _ zu viel und kein _ also zu wenige delimter, dann benenne die Variablen doch um nach apiKeyWithTooManyDelimiter und apiKeyWithTooLittleDelimiter dann wird die Absicht besser deutlich

Reply:
Alles klar. Klingt plausibel. Werde ich tun.

Reply:
Habe den Test etwas bearbeitet. Schau es dir gerne nochmal an.

Reply:
passt